File: Parcial1/Punto3a.r
#Created on Wed Feb 28 08:20:20 2018
#author: Juanda
#Punto 3a

# Remueve todos los objetos creados
rm(list=ls())
Fx <- function(x) log(x+2) - sin(x)
F1x <- function(x) (1/(x+2)) -cos(x)

# Metodo de la Secante
# Halla la raiz de Fx
secante <- function(x0,x1) {
  #(Fx(b)*a-Fx(a)*b)/(Fx(b)-Fx(a))
  x<-(Fx(x1)*x0-Fx(x0)*x1)/(Fx(x1)-Fx(x0))
  error <-1 #error se inicializa en 1
  while (error > 1.e-7) {
    x0<-x1 #ahora límite inferior igual a superior
    x1<-x #ahora límite superior igual a x calculado
    x<-(Fx(x1)*x0-Fx(x0)*x1)/(Fx(x1)-Fx(x0))
    if (Fx(x) == 0) # se encontró la raíz
      break
    error<-abs(Fx(x)/F1x(x))
    cat("X=",x,"\t","E=",error,"\n")
  }
}

system.time({secante(-1.8,-1.0)})

File: Parcial1/Punto3a.py
# -*- coding: utf-8 -*-
"""
Created on Wed Feb 28 07:42:16 2018

@author: Juanda
"""
import numpy as np

def f(x): return np.log(x+2) - np.sin(x)
def f1x(x): return (1/(x+2)) - np.cos(x)

def secante(x0,x1,tol):     
  
  x = (f(x1)*x0-f(x0)*x1)/(f(x1)-f(x0))
  error = 1 
  print("\t x","\t\t\t","Error")
  while (error > tol):      
    x0 = x1
    x1 = x
    x = (f(x1)*x0-f(x0)*x1)/(f(x1)-f(x0))
    if (f(x) == 0):
        break
    error = abs(f(x)/f1x(x))
    print(x,"\t",error)
  
secante(-1.8,-1.0,1.e-7)